src/trade/utils.ts

```diff
@@ -28,7 +10,7 @@ export function areCurrenciesEqual(
   if (currency.chainId !== chainId) return false;
 
   if (currency.isNative) {
-    return address === nativeCurrencyAddressString(chainId)
+    return address === NATIVE_ADDRESS;
```

Comment:
From the URA perspective we are setting the token address for native swaps to the zero address. Assuming this is okay to do for all chains?